agent-codeless/src/main/java/com/microsoft/applicationinsights/agentc/internal/diagnostics/PidFinder.java

```diff
@@ -0,0 +1,39 @@
+package com.microsoft.applicationinsights.agentc.internal.diagnostics;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+
+public class PidFinder extends CachedDiagnosticsValueFinder {
+    public static final String PROPERTY_NAME = "PID";
+
+    @Nullable
+    @Override
+    protected String populateValue() {
+        // will only work with sun based jvm
+        final RuntimeMXBean rb = ManagementFactory.getRuntimeMXBean();
```

Comment:
This looks fragile.
Can we use Java 9 API for this? ProcessHandle.current().pid();
and as a fallback (in case we need to support older jvms) - use the random string.

Reply:
Since we need to maintain Java 7/8 compatibility, I can't use it directly. I could call it via reflection, but this will work for oracle/openjdk based jvms and in app services, this will always be the case.

Reply:
We plan to re-use this code outside of AppSvcs, where any jvm could be present. Let's have a following logic: java9+ - reflection, else - the current code, else - random string?

Reply:
random string?